hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java

```diff
@@ -251,14 +253,28 @@ private static boolean checkModule(final String fileName, final SoftwareModule m
 
             final HandlingType handlingType = action.isForce() ? HandlingType.FORCED : HandlingType.ATTEMPT;
 
-            final DdiDeploymentBase base = new DdiDeploymentBase(Long.toString(action.getId()),
-                    new DdiDeployment(handlingType, handlingType, chunks));
+            DdiPreviousActionFeedback previousActionFeedback = null;
+            Optional<Action> actionWithDetails = controllerManagement.findActionWithDetails(actionId);
+
+            if (actionWithDetails.isPresent()) {
+                List<String> feedbackMessages = new ArrayList<>();
 
-            LOG.debug("Found an active UpdateAction for target {}. returning deyploment: {}", controllerId, base);
+                actionWithDetails.get().getActionStatus().stream()
+                        .forEach(actionStatus -> feedbackMessages.addAll(actionStatus.getMessages()));
```

Comment:
I think using `flatMap` instead for forach->addAll would make things easier readable.